### DISCUSSION TITLE
How can I get SSL certificate server of the server

### DISCUSSION THREAD

**Discussion by huenemeca** [micropython/micropython#12896] [Libraries & Drivers]
Hello,
I am working on a project where I need to use the modulus and exponent of the servers SSL certificate of the server.  Is there a way to get the ssl certificate from a server in micropython?

Thanks

**Comment by sophiedegran** [comment]
You have to do that on machine with openssl installed. (by default on linux it should be)
''' openssl req -x509 -newkey rsa:4096 -nodes -out cert.pem -keyout key.pem -days 365 '''
(see options of openssl...)
After that you have to convert PEM files to DER files for micropython with that command:
''' openssl x509 -in cert.pem -out cert.der -outform DER '''
''' openssl rsa -in key.pem -out key.der -outform DER '''
Then you can copy the 2 DER files to your micropython web server...
Remenber that for instance ssl is only implemented on sync mode under micropython
